x-pack/plugins/session_view/public/components/session_view/index.tsx

```diff
@@ -118,12 +118,34 @@ export const SessionView = ({
     hasPreviousPage,
   } = useFetchSessionViewProcessEvents(sessionEntityId, currentJumpToCursor);
 
-  const alertsQuery = useFetchSessionViewAlerts(sessionEntityId);
-  const { data: alerts, error: alertsError, isFetching: alertsFetching } = alertsQuery;
+  const {
+    data: alertsData,
+    fetchNextPage: fetchNextPageAlerts,
+    isFetching: isFetchingAlerts,
+    hasNextPage: hasNextPageAlerts,
+    error: alertsError,
+  } = useFetchSessionViewAlerts(sessionEntityId, investigatedAlertId);
+
+  const alerts = useMemo(() => {
+    let events: ProcessEvent[] = [];
+
+    if (alertsData) {
+      alertsData.pages.forEach((page) => {
+        events = events.concat(page.events);
+      });
+    }
+
+    return events;
+  }, [alertsData]);
+
+  const alertsCount = useMemo(() => {
+    return alertsData?.pages?.[0].total || 0;
```

Comment:
Probably (Hopefully) will never happen 🤞 but will this total also become negative when it gets too large? We could do `Math.max(alertsData?.pages?.[0].total || 0, 0)` but might be an overkill.

Reply:
For some reason the alerts indices were reporting an accurate total. I've made a note about this in the ticket tracking this issue. Haven't looked into why that is though. I've seen a total > 35000 on my local setup.

Reply:
> also, total will never return a negative value, it is just the calculation of substracting the total events loaded - total that resulted in a negative number (as the process indices don't ever seem to report a 'total' > 10000)

ah yes thats right.

> For some reason the alerts indices were reporting an accurate total. I've made a note about this in the ticket tracking this issue. Haven't looked into why that is though. I've seen a total > 35000 on my local setup.

I left a comment in the same ticket https://github.com/elastic/kibana/issues/130919#issuecomment-1122585578 just for information, it seems like no changes are required.

Reply:
Cool, yea if we hit the ceiling on this with alerts incide, it's not a big deal as we never show a total > 500 in the tab, and no "remaining count" for the load more button.